src/ingestion/sources/bigquery.py

```diff
@@ -0,0 +1,200 @@
+                COUNT(*) as `ROW_COUNT`, 
+                '{table}' as `TABLE_NAME`,
+                '{database}' as `DATABASE_NAME`,
+                '{schema}' as `SCHEMA_NAME`,
+
+                {select_sql}
+            FROM {schema}.{table};
+        """.format(
+            select_sql=select_sql,
+            table=self.monitor['table_name'],
+            schema=self.monitor['schema'],
+            database=self.monitor['database'],
+        )
+
+    def _base_query_backfill(self, select_sql, table, timestamp_field):
+        return """
+            SELECT 
+                DATE_TRUNC({timestamp_field}, HOUR) as `WINDOW_START`, 
+                TIMESTAMP_ADD(DATE_TRUNC({timestamp_field}, HOUR), INTERVAL 1 HOUR) as `WINDOW_END`, 
+                COUNT(*) as `ROW_COUNT`, 
+                {table} as `TABLE_NAME`,
+                {database} as `DATABASE_NAME`,
+                {schema} as `SCHEMA_NAME`,
+                {timestamp_field} as `TIMESTAMP_FIELD`,
+
+                {select_sql}
+            FROM {schema}.{table} as c
+            WHERE 
+                DATE_TRUNC({timestamp_field}, HOUR) >= TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL {minutes_ago} MINUTE)
+            GROUP BY `WINDOW_START`, `WINDOW_END`
+            ORDER BY `WINDOW_START` ASC;
+        """.format(
+            select_sql=select_sql,
+            table=table,
+            timestamp_field=timestamp_field,
+            minutes_ago=self.minutes_ago,
+            database=self.monitor['database'],
+            schema=self.monitor['schema'],
+        )
+
+class BigQuerySourceConfiguration(SourceConfiguration):
+    @classmethod
+    def validate(cls, configuration):
+        raise NotImplementedError
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "project": { "type": "string" },
+                "dataset": { "type": "string" },
+                "credentials_base64": { "type": "string" },
+            },
+        }
+    
+    def _connection_string_prefix(self):
+        return "bigquery"
+
+    def connection_string(self) -> str:
+        configuration = json.loads(self.configuration)
+        connection_string_prefix = self._connection_string_prefix()
+        connection_string_base = '{project}/{dataset}'.format(
+            project=configuration.get('project'),
+            dataset=configuration.get('dataset'),
+        )
+        connection_string_params = '?credentials_base64={credentials_base64}'.format(
+            credentials_base64=configuration.get('credentials_base64')
+        )
+
+        connection_string = '{connection_string_prefix}://{connection_string_base}{connection_string_params}'.format(
+            connection_string_prefix=connection_string_prefix,
+            connection_string_base=connection_string_base,
+            connection_string_params=connection_string_params
+        )
+
+        return make_url(connection_string)
```

Comment:
what is make_url doing here?

Also why not just use https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.engine.URL.create

If you're already using a sqlalchemy helper and the configuration of the string looks really hard to follow, it might be a bit more elegant

Reply:
Using `make_url()` for sanity to create a proper URL for the SQLAlchemy engine. 

I don't believe we can use `URL.create` because the parameters for creating a BQ connection string are not standard

Reply:
Disagree, but not going to block